### ISSUE TITLE
Currency Issues

### ISSUE THREAD

**Issue by belphegorspn** [nulli0n/ExcellentShop-spigot#14]
Hello, I've recently noticed the many issues I've encountered with pricing and currencies with the plugin in each of its module.

Here is a summary:

1. Virtual Shop - Creating a product seems to default to random currencies like gold even when in the settings it's set that the default currency is Vault. Disabling gold from the currency folders will make the items in your virtual shop that were still using gold disappear. You would need to enable gold again from your `/custom_item/gold.yml` and then you can manually change the currency of each product to use vault.

Error code: https://paste.gg/p/anonymous/cce8719938f74b72bf83f886a0e91fe6
![image](https://github.com/nulli0n/ExcellentShop-spigot/assets/66054833/dbfb0f47-c4ef-4cbc-a773-3b04593428dd)

2. Auction House - You cannot change the item's material if it's using CoinsEngine. Unlike `vault.yml` in `/plugins/ExcellentShop/currency/default` you can change its material because it has a format like this:

```
Enabled: true
Name: $
Format: '%currency_name%%price%'
Icon:
  Material: SUNFLOWER
  Amount: 1
  Durability: 0
  Name: '&e%currency_name%'
  Item_Flags: []
  Unbreakable: false
  ```
 There's no custom currency listed in `/plugins/ExcellentShop/currency/default` from coins-engine instead you would have to go to `/plugins/CoinsEngine/currencies` and edit your currency there but there's something different.
  
  ```
  Name: '#FFD700Coins'
Symbol: '🪙'
Format: '%currency_symbol%%amount%'
Command_Aliases: 'coins,coin,forged'
Decimal: false
Permission_Required: false
Transfer_Allowed: true
Start_Value: 0
Max_Value: -1
Economy:
  Vault: false
  ```
 It lacks the `Icon` configuration from `/plugins/ExcellentShop/currency/default`
  
  
  Here's a full startup log if needed: https://mclo.gs/Wr4unAD

**Comment by nulli0n** [comment]
1. Should be fixed in v4.5.0

**Comment by nulli0n** [comment]
2. Finally fixed in the latest CoinsEngine.
